File: idakonyvtar/src/main/java/eu/eyan/idakonyvtar/util/SpecialCharacterRowFilter.java
package eu.eyan.idakonyvtar.util;

import java.util.regex.Matcher;

import org.jdesktop.swingx.sort.RowFilters.GeneralFilter;

public class SpecialCharacterRowFilter extends GeneralFilter
{
    private Matcher matcher;

    public SpecialCharacterRowFilter(String filterText)
    {
        matcher = SpecialCharacterHelper.filterPattern(filterText).matcher("");
    }

    @Override
    protected boolean include(javax.swing.RowFilter.Entry<? extends Object, ? extends Object> value, int index)
    {
        matcher.reset(SpecialCharacterHelper.withoutSpecChars(value.getStringValue(index)));
        return matcher.find();
    }
}

File: idakonyvtar/src/main/scala/eu/eyan/idakonyvtar/util/SpecialCharacterRowFilter.scala
package eu.eyan.idakonyvtar.util;

import java.util.regex.Matcher;

import org.jdesktop.swingx.sort.RowFilters.GeneralFilter;

class SpecialCharacterRowFilter(filterText: String) extends GeneralFilter {
  val matcher = SpecialCharacterHelper.filterPattern(filterText).matcher("")

  def include(value: javax.swing.RowFilter.Entry[_, _], index: Int): Boolean = {
    matcher.reset(SpecialCharacterHelper.withoutSpecChars(value.getStringValue(index)))
    matcher.find()
  }
}